### PR TITLE
Add DYNAMIC_ROUTER settings

### PR DESCRIPTION
**Pull request by Thib17**
cc @iksaif

### CODE REVIEW

**Comment by drawks** [comment]
Interesting features, but I think it would be more appropriate to split them into multiple PRs. I don't see any intrinsic overlap of function or code that should make these a single PR.

**Comment by Thib17** [comment]
Ok, so let's start with DYNAMIC_ROUTER settings first.

File: lib/carbon/client.py [hunk 1]
```diff
@@ -306,20 +315,69 @@ def sendHighPriorityDatapoint(self, metric, datapoint):
...
+      log.clients("Adding client %s:%d:%s to router" % destination)
+      self.router.addDestination(destination)
+      state.events.resumeReceivingMetrics()
+
+  def destinationDown(self, destination):
+    # Only blacklist the destination if we tried a lot.
+    log.clients("Destination is down: %s:%d:%s (%d/%d)" % (
+        destination[0], destination[1], destination[2], self.retries,
+        settings.DYNAMIC_ROUTER_MAX_RETRIES))
+    # Retries com from the ReconnectingClientFactory.
```
Comment (deniszh) [inline lib/carbon/client.py]: com?

**Comment by deniszh** [comment]
Interesting feature, indeed! Thanks, @Thib17 !
More reviews needed. /cc @drawks @iksaif @DanCech @graphite-project/committers

File: conf/carbon.conf.example [hunk 2]
```diff
@@ -370,6 +370,17 @@ DESTINATIONS = 127.0.0.1:2004
 # extended with CarbonClientFactory plugins and defaults to "pickle".
 # DESTINATION_PROTOCOL = pickle
 
+# When using consistent hashing it sometime make sense to make
```
Comment (iksaif) [inline conf/carbon.conf.example]: makes*

File: lib/carbon/client.py (hunk 1 above)
Comment (iksaif) [inline lib/carbon/client.py]: comes*

File: conf/carbon.conf.example (hunk 2 above)
Comment (Thib17) [inline conf/carbon.conf.example]: Done

File: lib/carbon/client.py (hunk 1 above)
Comment (Thib17) [inline lib/carbon/client.py]: Done

**Pull request merged**